tests/io/test.directory.js

```diff
@@ -61,85 +55,86 @@ describe('Directory.getFiles()', () => {
-      .then(() => {
-        return myDirectory.getPath('whatever')
-          .then(unexpectedSuccess)
-          .catch((err) => {
-            expect(err.message).toContain(
-              '"whatever" does not exist in this dir.'
-            );
-          });
-      });
+
+    await myDirectory.getFiles();
+    try {
+      await myDirectory.getPath('whatever');
+      unexpectedSuccess();
+    } catch (err) {
+      expect(err.message).toContain(
+        '"whatever" does not exist in this dir.'
+      );
+    }
   });
 
-  it('should reject if path does not start with base', () => {
+  it('should reject if path does not start with base', async () => {
     const myDirectory = new Directory('tests/fixtures/io/');
     myDirectory.files = {
       '../file1.txt': {},
     };
-    return myDirectory.getPath('../file1.txt')
-      .then(unexpectedSuccess)
-      .catch((err) => {
-        expect(err.message).toContain('Path argument must be relative');
-      });
+
+    try {
+      await myDirectory.getPath('../file1.txt');
+      unexpectedSuccess();
+    } catch (err) {
+      expect(err.message).toContain('Path argument must be relative');
+    }
   });
 
-  it("should reject if path starts with '/'", () => {
+  it("should reject if path starts with '/'", async () => {
     const myDirectory = new Directory('tests/fixtures/io/');
     myDirectory.files = {
       '/file1.txt': {},
     };
-    return myDirectory.getPath('/file1.txt')
-      .then(unexpectedSuccess)
-      .catch((err) => {
-        expect(err.message).toContain('Path argument must be relative');
-      });
+
+    try {
+      await myDirectory.getPath('/file1.txt');
+      unexpectedSuccess();
+    } catch (err) {
+      expect(err.message).toContain('Path argument must be relative');
+    }
   });
 });
 
 describe('Directory.getFileAsStream()', () => {
-  it('should return a stream', () => {
+  it('should return a stream', async () => {
     const myDirectory = new Directory('tests/fixtures/io/');
-    return myDirectory.getFiles()
-      .then(() => {
-        return myDirectory.getFileAsStream('dir2/dir3/file3.txt');
-      })
-      .then((readStream) => {
-        return new Promise((resolve, reject) => {
-          let content = '';
-          readStream
-            .on('readable', () => {
-              let chunk;
-              // eslint-disable-next-line no-cond-assign
-              while ((chunk = readStream.read()) !== null) {
-                content += chunk.toString();
-              }
-            })
-            .on('end', () => {
-              resolve(content);
-            })
-            .on('error', (err) => {
-              reject(err);
-            });
-        })
-          .then((content) => {
-            expect(content).toEqual('123\n');
-          });
+    await myDirectory.getFiles();
+
+    const readStream = await myDirectory.getFileAsStream('dir2/dir3/file3.txt');
+
+    const readStringFromStream = () => {
```

Comment:
Nit, there is no need to declare it as a function if we are not going to reuse it,
we could just store the promise:

```js
const onceReadString = new Promise((resolve, reject) => {
  ...
});

const content = await onceReadString;
...
```

or we could move the function outside of the test function and pass the readStream as a parameter:

```js
function readStringFromStream(readStream) {
  return new Promise((resolve, reject) => {
    ...
  });
}

describe('...', () => {
  it('...', async () => {
    ...
    const content = await readStringFromStream(readStream);
    ...
  });
});
```

Reply:
@rpl I decided to go with second option because readStringFromStream(readStream) is used in 2 tests